packages/ai/core/tool/mcp/types.ts

```diff
@@ -0,0 +1,269 @@
+            options: ToolExecutionOptions,
+          ) => PromiseLike<CallToolResult>;
+        };
+      }
+    : {
+        [k: string]: Tool<z.ZodUnknown, CallToolResult> & {
+          execute: (
+            args: unknown,
+            options: ToolExecutionOptions,
+          ) => PromiseLike<CallToolResult>;
+        };
+      };
+
+export interface McpStdioServerConfig {
+  command: string;
+  args?: string[];
+  env?: Record<string, string>;
+  stderr?: IOType | Stream | number;
+  cwd?: string;
+  type: 'stdio';
+}
+export interface McpSSEServerConfig {
+  type: 'sse';
+  url: string;
+}
+export type TransportConfig = McpStdioServerConfig | McpSSEServerConfig;
+
+const ClientOrServerImplementationSchema = z
+  .object({
+    name: z.string(),
+    version: z.string(),
+  })
+  .passthrough();
+export type Configuration = z.infer<typeof ClientOrServerImplementationSchema>;
+
+const BaseParamsSchema = z
+  .object({
+    _meta: z.optional(z.object({}).passthrough()),
+  })
+  .passthrough();
+type BaseParams = z.infer<typeof BaseParamsSchema>;
+
+export interface Request {
+  method: string;
+  params?: BaseParams;
+}
+
+export type RequestOptions = {
+  signal?: AbortSignal;
+  timeout?: number;
+  maxTotalTimeout?: number;
+};
+
+const RequestSchema = z.object({
+  method: z.string(),
+  params: z.optional(BaseParamsSchema),
+});
+const ResultSchema = BaseParamsSchema;
+const NotificationSchema = z.object({
+  method: z.string(),
+  params: z.optional(BaseParamsSchema),
+});
+
+export type Notification = z.infer<typeof NotificationSchema>;
+
+const RequestIdSchema = z.union([z.string(), z.number().int()]);
+const JSONRPCRequestSchema = z
+  .object({
+    jsonrpc: z.literal(JSONRPC_VERSION),
+    id: RequestIdSchema,
+  })
+  .merge(RequestSchema)
+  .strict();
+export type JSONRPCRequest = z.infer<typeof JSONRPCRequestSchema>;
+const JSONRPCResponseSchema = z
+  .object({
+    jsonrpc: z.literal(JSONRPC_VERSION),
+    id: RequestIdSchema,
+    result: ResultSchema,
+  })
+  .strict();
+export type JSONRPCResponse = z.infer<typeof JSONRPCResponseSchema>;
+const JSONRPCErrorSchema = z
+  .object({
+    jsonrpc: z.literal(JSONRPC_VERSION),
+    id: RequestIdSchema,
+    error: z.object({
+      code: z.number().int(),
+      message: z.string(),
+      data: z.optional(z.unknown()),
+    }),
+  })
+  .strict();
+export type JSONRPCError = z.infer<typeof JSONRPCErrorSchema>;
+const JSONRPCNotificationSchema = z
+  .object({
+    jsonrpc: z.literal(JSONRPC_VERSION),
+  })
+  .merge(NotificationSchema)
+  .strict();
+export type JSONRPCNotification = z.infer<typeof JSONRPCNotificationSchema>;
+
+export const JSONRPCMessageSchema = z.union([
+  JSONRPCRequestSchema,
+  JSONRPCNotificationSchema,
+  JSONRPCResponseSchema,
+  JSONRPCErrorSchema,
+]);
+export type JSONRPCMessage = z.infer<typeof JSONRPCMessageSchema>;
+
+export interface MCPTransport {
+  start(): Promise<void>;
+  send(message: JSONRPCMessage): Promise<void>;
```

Comment:
do we need zod schemas for all of this (as in, we use the schemas), or can we go more lightweight and just define TS types (at least for some)?